app/src/main/java/hr/foi/air/servicesync/backend/FirestoreCompanyDetails.kt

```diff
@@ -47,4 +53,32 @@ class FirestoreCompanyDetails {
         }
     }
 
+    fun loadCompanyOpeningTimeByName(companyName: String, onResult: (String?) -> Unit) {
+        db.collection("companies")
+            .whereEqualTo("name", companyName)
+            .limit(1)
+            .get()
+            .addOnSuccessListener { documents ->
+                val openingTime = documents.documents.firstOrNull()?.getString("openingTime")
+                companyOpeningTime = openingTime ?: "0:00"
+                onResult(companyOpeningTime)
+            }
+            .addOnFailureListener {
+                onResult("0:00")
+            }
+    }
+    fun loadCompanyClosingTimeByName(companyName: String, onResult: (String?) -> Unit) {
+        db.collection("companies")
+            .whereEqualTo("name", companyName)
+            .limit(1)
+            .get()
+            .addOnSuccessListener { documents ->
+                val closingTime = documents.documents.firstOrNull()?.getString("closingTime")
+                companyClosingTime = closingTime ?: "23:59"
+                onResult(companyClosingTime)
+            }
+            .addOnFailureListener {
+                onResult("23:59")
+            }
+    }
 }
```

Comment:
There should be a class Company that holds all info about the company whose details are being displayed. This is kinda redundant because multiple fetches happen, potentially could lead intense traffic. Definitely consider this in the upcoming sprints